### DISCUSSION TITLE
realism and accuracy

### DISCUSSION THREAD

**Discussion by axd1967** [Stellarium/stellarium#1504] [General]
As a general note, sometimes the arguments pops up that Stellarium wants to be "realistic", and some wishes are classified as foolish, physically impossible, etc.

One needs to remember that this is software that is capable to highlight phenomena that are not always well visible, but it should not block _a priori_ things that seem "physically impossible". Also, software is capable of producing things that will never be visible in reality, and this can provide insight.

A realistic mode is certainly nice to have, but often we need to amplify things, and sometimes exaggerate on purpose, eg for educational reasons, or simply for artistic reasons (eg think of the - indeed unrealistic - 300x telescope (#1178), the limits on FOV values (#1196), invisible comets, etc).

It is not because something is physically impossible, that Stellarium should systematically **enforce** that limit. As long as we steer away from e.g. "division by zero" situations, software should allow to see things that might not be evident in reality. Of course, Stellarium is not the most accurate application, but attempts to mask such imperfections should receive a lot of attention.

Because if one really wants to have Stellarium enforce physical limits, one should - just as an example, but  many more could be given - not allow the user to look below the horizon: that's just one simple example of how Stellarium allows more than is physically possible (e.g. if Stellarium would stick to "realistic" mode, it should force the user to change location in order to see what's happening below the horizon). Another example are the nice images one can make appear under the DSS/TOAST/HIPS buttons: these are totally unrealistic for most telescope users without image processing - that's another example of how Stellarium distorts reality for the sake of insight.

So in the issue of #1489 , allow for slack. Sometimes, for educational purposes, there is a need to exaggerate the ZL visual, and an educator will draw the attention that in reality ZL is much weaker. For example, exaggerating can help in better visualise how time and latitude can have impact on the orientation of ZL.

**Comment by gzotti** [comment]
That light pollution kills ZL was implemented with an educative intention. We can discuss fine-tuning (therefore the issue is not closed!), but it would be horrific nonsense to allow any setting that would show ZL in the middle of a bright city with Bortle index>7 without switching off the atmosphere. To visualize latitude dependence, you can exagerrate ZL brightness beyond the brightness I find roughly realistic. I would have to observe it again for further finetuning, but Atque may suggest better values earlier. 
GUI limits are there for a reason, as discussed in your various "issues". A 300x bino is plain nonsense from laws of optical physics (no nanotech will fix that!) and there is just no practical use for such high limits, but you can easily define an instrument (Telescope+Ocular) combination that will yield usable 300x. Simple FOV box vs. simulation of a sensor/telescope box: same thing. FOV indicator for an all-sky camera? I don't see the point, it's all-sky, right?
Invisible comets are out-of-visibility and just would reduce performance. There is a reason behind our decisions, but if you know everything better, demonstrate us how incredibly stupid we are and provide demonstrably better solutions. The nice thing with open-source is that you can build your private dream version with all limits removed. Load 100.000 asteroids in the current program and complain about frame rate breakdown in your next issue... Or do something helpful like develop a compute shader solution which shows points for each known object in the solar system.

**Reply by axd1967** [reply]
Please stop being so denigrating. I've always respected everybody and expect the same attitude.

I _don't_ know everything better, in some issues I have acknowledged I was wrong.

I _don't_ state anywhere anyone is incredibly stupid, and if I did, that must have been an exception

I _don't_ ask to show 100000 asteroids _just like that_. I _do_ know _very well_ it will affect frame rate.

I _know_ asking to show tons of comets will impact performance, but as I explain in that issue, that is a decision to make by the user, by means of options that are chosen so as to protect those users that do not use those options. A power user should not be blocked by weak users, but weak users should be protected from features that only power users can use. Again: the software should allow crazy things for someone who wants to try crazy things - they just should not complain if their hardware is not able to follow.

And the all-sky cam is NOT always 180 degrees open; there can be issues that limit the sky to slightly lower values, eg 170 degrees opening, and a 160 opening is also interesting, because it simulates a relatively open field with a 10-degree horizon; how about that?

Please re-read https://github.com/Stellarium/stellarium/issues/1484#issuecomment-768887843

Let's focus on what can be done now, not on what _cannot_ be done, as the future might provide solutions to problems that _seem_ impossible to solve today.

Nobody asks to break physical laws;  but look - Stellarium DOES allow to simulate light speed! that is NOT realistic, is it? That IS breaking the laws of physics, yet I hear nobody complain.

So in this ZL case, and more in general, **the essence is that the average user has a means to know what the realistic values** are, but also the means to exaggerate it, _just for fun, for artistic purposes_! For the software it makes no difference, as long as over/underflows don't break the experience. Experienced users can suspect that in the case of visibility many parameters come into play, and Stellarium is not 100% correct, but again, **the point is: allow for slack**.

(extreme ZL with atmosphere, but only around sunrise/set:) 

![image](https://user-images.githubusercontent.com/3529789/107941875-b6531400-6f8a-11eb-9afd-6417a2ec29ce.png)

![image](https://user-images.githubusercontent.com/3529789/107943288-ca981080-6f8c-11eb-943c-77dcd32e93b0.png)

![image](https://user-images.githubusercontent.com/3529789/107943412-f7e4be80-6f8c-11eb-8a1b-f9ec04c53d1d.png)

(extreme ZL without atmosphere: is this realistic? I don't know, we should ask astronauts for that...)

![image](https://user-images.githubusercontent.com/3529789/107941948-d08cf200-6f8a-11eb-911c-ecde768ec9b6.png)

**Reply by gzotti** [reply]
ZL (and in the same way, Milky Way) brightness of "1" is what "should" look realistic, but it may depend on the brightness around your monitor to see it well, so you may want to exaggerate its brightness. If you exaggerate ZL to 10, it is not expected to be realistic but already grotesque, and shows technical limitations (whiteout). But if I follow your logic, we need to allow going to 1000 or more. How do other programs behave here?

**Reply by axd1967** [reply]
1=realistic is fine. 1000 is probably an exageration, as the screenshots already show enough (away from daylight, which sounds an acceptable limitation. the issue did not focus on daylight.)
but the ZL discussion should continue in the issue, while this discussion should focus on the general idea of realism?
